### PR TITLE
lz4: Fix fPIC unsafe remove in configure

### PR DESCRIPTION
**Pull request by RubenRBS**
Specify library name and version:  **lz4/all**

Additionally, it does some other minor changes:
- Also uses rm_safe for compiler.cppstd & compiler.libcxx
- Cleans its topics up
- Uses `export_conandata_patches`
- Adds info to its patches

### CODE REVIEW

File: recipes/lz4/all/conanfile.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ class LZ4Conan(ConanFile):
     license = ("BSD-2-Clause", "BSD-3-Clause")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/lz4/lz4"
-    topics = ("lz4", "compression")
```
Comment (SpaceIm) [inline recipes/lz4/all/conanfile.py:18]: I would not remove lz4 as a topic, lz4 is not only library name but also compression algorithm name.

File: recipes/lz4/all/conanfile.py [hunk 2]
```diff
@@ -28,24 +28,17 @@ class LZ4Conan(ConanFile):
...
-            del self.settings.compiler.libcxx
-        except Exception:
-            pass
-        try:
-            del self.settings.compiler.cppstd
-        except Exception:
-            pass
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
```
Comment (SpaceIm) [inline recipes/lz4/all/conanfile.py:41]: required_conan_version should be updated to `>=1.53.0`

File: recipes/lz4/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/lz4/all/conanfile.py:18]: the topics are for searching and the name automatically is included in that AFAIK so it should be fine

**Review by prince-chrismc — APPROVED** [review]

File: recipes/lz4/all/conanfile.py (hunk 2 above)
Comment (RubenRBS) [reply recipes/lz4/all/conanfile.py:41]: Oops, missed that change, I knew I was forgetting something! Thanks :)

File: recipes/lz4/all/conanfile.py [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
-from conan.tools.files import apply_conandata_patches, copy, get, rmdir, save
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir, save
 from conan.tools.microsoft import is_msvc
 from conan.tools.scm import Version
 import os
 import textwrap
 
-required_conan_version = ">=1.50.0"
+required_conan_version = ">=1.54.0"
```
Comment (SpaceIm) [inline recipes/lz4/all/conanfile.py:9]:
```suggestion
required_conan_version = ">=1.53.0"
```

Comment (SpaceIm) [reply recipes/lz4/all/conanfile.py:9]: But if you want to remove `tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW", yes 1.54.0 would be required.

**Review by prince-chrismc — APPROVED** [review]

File: recipes/lz4/all/conanfile.py (hunk 3 above)
Comment (RubenRBS) [reply recipes/lz4/all/conanfile.py:9]: Didn't think that far ahead for this PR :) As my knowledge is limited, I'm trying to change as little as I can, so I'll leave any proper changes to pros like you :)

**Review by jcar87 — APPROVED** [review]

**Pull request merged**